library/aes.c

```diff
@@ -655,6 +655,13 @@ int mbedtls_aes_setkey_enc(mbedtls_aes_context *ctx, const unsigned char *key,
     }
 #endif
 
+/* When runtime detection enabled and plain C is disabled, compiler
+   reports `-Werror=return-type`. */
+#if defined(MBEDTLS_HAVE_X86) && defined(MBEDTLS_AES_USE_HARDWARE_ONLY) && \
```

Comment:
doesn't `defined(MBEDTLS_AES_USE_HARDWARE_ONLY)` mean runtime detection is disabled? This seems inconsistent with the comment?

Reply:
For i386 case, there are two hardware implementations. So it does not mean runtime detection is disabled.

Reply:
because padlock code depends on plain c, this case is not reasonable. I will remove it.

Reply:
It has been removed